docs/pages/hdmi_cec_source_tests.md

```diff
@@ -0,0 +1,176 @@
+
+## Scope
+
+This document defines the requirements for testing the HDMI CEC device from a level 2/3 based on being a source device.
+
+## Overview
+
+Consumer Electronics Control (CEC) is a single-wire bidirectional bus within an HDMI system, facilitating communication among interconnected products. HDMI-CEC establishes a protocol enabling high-level control functions between audiovisual devices linked via an HDMI network, facilitating communication and control among them. Communication can occur in either Direct messaging mode or Broadcast mode. 
+
+## HDMI-CEC RDK HAL Functionality
+
+The HAL layers within RDK serve as a bridge between the underlying low-level SoC drivers and the higher-level RDK layers that utilize the functionality offered by these HAL functions. Specifically concerning the CEC Module, the HAL layers facilitate the following functionalities:
+
+- Get Logical address
+- Get Physical address
+- Syncronous transmission, and communicating via hotplug connectivity
+
+## Test scenarios
+
+The HAL CEC layer facilitates the transmission and reception of CEC information on the CEC bus. It does not handle any specific opcode commands, nor does it validate supported HAL CEC opcodes for sending or receiving.
+
+It is the responsibility of the caller to manage the opcodes. The current test cases will verify responses from connected devices for a subset of opcodes as part of the testing process.
+
+|S.No.|Test Functionality|Description|
+|-----|------------------|-----------|
+| 1 |[Logical address](#logical-address-discovery)|Facilitating the Discovery of logical addresses getting, and removing the logical address of the device (for sink devices) |
+| 2| [Physical Address](#physical-address)| Retrieving the physical address |
+| 3| [CEC Synchronous Transmission](#cec-synchronous-transmission)| Transmitting CEC frames and acknowledging them |
+| 4| [CEC Receive functionality](#cec-receive-functionality)| Receiving CEC information from other devices and communicating it to the above layers through registered callback functions |
+| 5| [CEC HotPlug Functionality](#cec-hotplug-functionality)| Managing CEC during Hotplug and HotUnplug events |
+
+-----------
+
+## Logical Address Discovery
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+| 1 |[Logical address](#logical-address-discovery)|Get the logical address of the `DUT`. This will add the logical address, as per source functionality. |HdmiCecGetLogicalAddress|Y|NA
+
+### Emulator Requirements
+
+- Boot with control configuration with various configurations having a predefined set of nodes:
+  - configuration to support the discovery of logical addresses. The caller provides the logical address, and HAL checks the availability of this address and feedback the same to the caller. 
```

Comment:
On Source devices, caller doesn't poll to get the logical address. It is the open function which should get the LA.  Emulator requirement should be saying "to provide a create a proper logical address during Open and that should be  provided when HDMICECGetLogicaladdress is used."

Reply:
Updated as requested